### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by hovsepm**
- Fixed issues with listing subscriptions for non-interactive and tenant-bound credentials
- Fixed ServicePrincipal login-azurermaccount issue related to ParameterSet mismatch 
- Added -EnvironmentName parameter in Login-AzureRmAccount to accept environment name string instead of environment

### CODE REVIEW

**Comment by hovsepm** [comment]
on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/343/

**Comment by hovsepm** [comment]
on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/347/

File: src/ResourceManager/Profile/Commands.Profile/Models/ModelExtensions.cs [hunk 1]
```diff
@@ -31,5 +35,27 @@ internal static AzureSubscription ToAzureSubscription(this Subscription other, A
                 context.Tenant.Id.ToString());
             return subscription;
         }
+
+        public static List<AzureTenant> MergeTenants(
```
Comment (stankovski) [inline src/ResourceManager/Profile/Commands.Profile/Models/ModelExtensions.cs]: This method name is a bit misleading - we are not merging tenants here - we are setting/overriding them

**Comment by stankovski** [comment]
LGTM

**Pull request merged**